clients/wallet/src/resubmissions.rs

```diff
@@ -193,6 +193,83 @@ impl StellarWallet {
 	}
 }
 
+fn is_source_account_match(public_key: &PublicKey, tx: &TransactionResponse) -> bool {
+	match tx.source_account() {
+		Err(_) => false,
+		Ok(source_account) if !source_account.eq(&public_key) => false,
+		_ => true,
+	}
+}
+
+fn is_memo_match(tx1: &Transaction, tx2: &TransactionResponse) -> bool {
+	if let Some(response_memo) = tx2.memo_text() {
+		let Memo::MemoText(tx_memo) = &tx1.memo else {
+			return false
+		};
+
+		if are_memos_eq(response_memo, tx_memo.get_vec()) {
+			return true
+		}
+	}
+	false
+}
+
+#[doc(hidden)]
+/// A helper function which:
+/// Returns true if the transaction in the middle of the list is a match;
+/// Returns false if NOT a match;
+/// Returns None if transaction can be found else where by narrowing down the search:
+///  * if the sequence number of the transaction is > than what we're looking for, then update the
+///    iterator by removing the first half of the list;
+///  * else remove the last half of the list
+///
+/// # Arguments
+///
+/// * `iter` - the iterator to iterate over a list of `TransactionResponse`
+/// * `tx` - the transaction we want to confirm if it's already submitted
+/// * `public_key` - the public key of the wallet
+fn _is_transaction_already_submitted(
```

Comment:
Some checks in this function are duplicated with `is_transaction_already_submitted&#40;&#41;`. Wouldn't it be enough to have these checks in one of the two functions?

Reply:
got it in 1 function: ` fn _check_transaction_match(...)` though the return datatype is quite confusing, so I added comments on the function; 
including the ff functions: 
* `fn check_last_transaction_match(...)`
* `check_middle_transaction_match(...)`

It is is some kind of binary search; but the checking of the last element plays a role: whether the transaction can be found on the next page or not.